addon/-private/create-reducer-helper.js

```diff
@@ -0,0 +1,41 @@
+import Ember from 'ember';
+
+const {
+  Helper,
+  K,
+  computed,
+  observer,
+  set,
+  get
+} = Ember;
+
+/**
+ * Creates a generic Helper class implementation that expects a `needle` and
+ * `haystack` as arguments. A `reducer` function is required to be passed in
+ * that reduces the `haystack` down into a single value.
```

Comment:
I would say this file doesn't actually have anything to do with reduce. It actually just creates a helper that takes two array arguments, and updates when their contents change. The two times you use it, you happen to use a reducing function. It can be left as is, but it's a little confusing because the names introduce a concept that doesn't exist in the code. I was confused and looking around for a `reduce` call